### PR TITLE
DTSBPS-444: Adding payment controller

### PR DESCRIPTION
**Pull request by SomeshRA**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DTSBPS-444

### Change description ###
Adding payment controller

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/PaymentController.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        @ApiResponse(code = 200, message = SUCCESSFUL_UPATE),
+        @ApiResponse(code = 401, message = "Invalid service authorisation header"),
+        @ApiResponse(code = 403, message = "Service not configured"),
+        @ApiResponse(code = 400, message = "Bad request")
+    })
+    public ResponseEntity<String> updatePayemnts(
+        @RequestHeader(name = "ServiceAuthorization", required = false) String serviceAuthHeader,
+        @Valid @RequestBody PaymentRequest paymentRequest) {
+
+        authService.authenticate(serviceAuthHeader);
```
Comment (mdayican) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/PaymentController.java:44]: we can log the DCNs

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/PaymentControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    void should_return_status_400_for_payment_with_empty_dcn() throws Exception {
+        String request = TestUtil.fileContentAsString(PAYMENT_DCN_EMPTY_JSON);
+
+        //When
+        mockMvc.perform(put("/payment/status")
+                            .contentType(MediaType.APPLICATION_JSON)
+                            .header("ServiceAuthorization", "testServiceAuthHeader")
+                            .content(request))
+            .andDo(print())
+            .andExpect(status().isBadRequest());
```
Comment (mdayican) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/PaymentControllerTest.java:110]: better to test the `field` error values

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/PaymentControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+@AutoConfigureMockMvc
+@IntegrationTest
+public class PaymentControllerTest {
+    private static final String PAYMENT_REQUEST_JSON = "testData/payment/payments.json";
+    private static final String PAYMENTS_EMPTY_JSON = "testData/payment/payments_empty.json";
+    private static final String PAYMENT_DCN_EMPTY_JSON = "testData/payment/payments_dcn_empty.json";
+
+    @Autowired
+    private MockMvc mockMvc;
+
```
Comment (mdayican) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/PaymentControllerTest.java:39]: extra line

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/PaymentController.java (hunk 1 above)
Comment (SomeshRA) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/PaymentController.java:44]: DCNs are logged in the service after performing all validations.

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/PaymentControllerTest.java (hunk 2 above)
Comment (SomeshRA) [reply src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/PaymentControllerTest.java]: Verified content.

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/PaymentControllerTest.java (hunk 3 above)
Comment (SomeshRA) [reply src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/PaymentControllerTest.java:40]: Removed extra line

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/PaymentControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+            new Payment("11234"),
+            new Payment("22234"),
+            new Payment("33234")
+        );
+
+        //Given
+        Envelope envelope = EnvelopeCreator.envelope(payments);
+        envelopeRepository.save(envelope);
+
+        List<PaymentInfo> paymentInfoList = of(
```
Comment (apereverzin) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/PaymentControllerTest.java:59]: Unused

**Review by apereverzin — APPROVED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/PaymentControllerTest.java (hunk 4 above)
Comment (SomeshRA) [reply src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/PaymentControllerTest.java]: Removed this list.

**Pull request merged**